### PR TITLE
chore: wei to ether

### LINKED ISSUE #172: feat: decimal conversion util function
#### Issue:
Currently, our web3 util tools lack a function that facilitates the conversion of a given amount in Wei to Ether, considering a specified number of decimals. This addition would enhance usability and provide a straightforward method for this common conversion.

#### Proposal:
I propose adding a new function to our web3 utilities that accepts two arguments: an amount in Wei and a decimal value. This function would convert the Wei amount to Ether, applying the appropriate decimal precision, and return the result as a string.

#### Implementation Details:
- Define a function in our web3 utilities to perform the Wei to Ether conversion.
- The function should accept a Wei amount and a decimal count as parameters.
- Return the result as a string to maintain consistency.

#### Testing:
To ensure the accuracy of the conversion function:
- Add comprehensive unit tests that cover various scenarios (different Wei amounts and decimal inputs).
- Validate the correctness of the converted Ether values against expected outputs.

#### Benefits:
- Improved usability by providing a direct method for Wei to Ether conversion.
- Enhanced reliability and confidence through comprehensive unit tests.

For example, this can be used in the following context and many others in the future:

https://github.com/grindery-io/grindery-bot-auth/blob/377c036e82ec00738c76849538fa6cce534f9ef1/src/utils/swap.ts#L198-L218

### PR DESCRIPTION
**Pull request by juniormp**
Close #172.

### CODE REVIEW

File: src/test/utils-web3.test.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import chai from 'chai';
-import { numberToString, scaleDecimals } from '../utils/web3';
+import { numberToString, scaleDecimals, weiToEther } from '../utils/web3';
+import BN from 'bn.js';
```
Comment (tcoratger) [inline src/test/utils-web3.test.ts:3]: In the other file you use the other lib: `import BigNumber from 'bignumber.js';` can we use `bignumber.js` here too for consistency? (This is in my TODO list to open an issue to use a single big number lib everywhere and remove the other one).

**Comment by tcoratger** [comment]
I've just left a small comment, other than that, seems good to me. Also, now that we have this util, can you call it here to replace the old lines of code: https://github.com/grindery-io/grindery-bot-auth/blob/377c036e82ec00738c76849538fa6cce534f9ef1/src/utils/swap.ts#L198-L218

File: src/test/utils-web3.test.ts (hunk 1 above)
Comment (juniormp) [reply src/test/utils-web3.test.ts:3]: @tcoratger using bignumber.js the result is not that precise comparing to BigNumber

Comment (tcoratger) [reply src/test/utils-web3.test.ts:3]: Then let us use the other one, but better if this is the same everywhere (meaning for tests and function)

File: src/webhooks/swap.ts [hunk 2]
```diff
@@ -283,16 +283,14 @@ export class SwapTelegram {
...
-        .div(
-          10 **
-            (nativeTokenAddresses.includes(this.params.tokenIn)
-              ? 18
-              : await getContract(this.params.chainIn, this.params.tokenIn)
-                  .methods.decimals()
-                  .call()),
-        )
-        .toString(),
+      amountIn: weiToEther(
```
Comment (tcoratger) [inline src/webhooks/swap.ts:286]: Same for `amountOut`

**Pull request merged**